src/group/organizationGroup/organizationGroup.textSearch.ts

```diff
@@ -0,0 +1,14 @@
+import { IOrganizationGroup } from './organizationGroup.interface';
+
+export type GroupQuery = {
+  nameAndHierarchyTerms: string | string[];
+};
+
+export type GroupFilters = {
+  hierarchyPath: string;
+};
+
+export interface OrganizationGroupTextSearch {
+  searchByNameAndHierarchy(query: Partial<GroupQuery>, filters?: Partial<GroupFilters>)
+    : Promise<IOrganizationGroup[]>;
+}
```

Comment:
maybe refactor to `searchByNameAndHierarchy(nameAndHierarchyTerms, filter)` and make `nameAndHierarchyTerms` required